api/client/webclient/webclient.go

```diff
@@ -47,14 +50,45 @@ func newWebClient(insecure bool, pool *x509.CertPool) *http.Client {
 	}
 }
 
+// getWithFallback attempts to execute an HTTP GET using https, and then fall
+// back to plain HTTP under certain, very specific circumstances.
+//  * The caller must specifically allow it via the allowPlainHTTP parameter, and
+//  * The target host must resolve to the loopback address.
+// If these conditions are not met, then the plain-HTTP fallback is not allowed,
+// and a the HTTPS failure will be considered final.
+func getWithFallback(ctx context.Context, clt *http.Client, allowPlainHTTP bool, proxyAddr string, path string) (*http.Response, error) {
+	// first try https and see how that goes
+	endpoint := fmt.Sprintf("https://%s%s", proxyAddr, path)
+	log.Debugf("Attempting %s", endpoint)
```

Comment:
nit: one of the goals of the new API is to limit unnecessary dependencies. Would basic `log` work here instead of logrus? Ideally we will remove the indirect dependency from `trace` at some point as well.

Reply:
Can definitely use plain ol' log. Logrus is till used by other parts of the `api` code though.

Reply:
Removed